src/Security/UserClassBuilder.php

```diff
@@ -22,21 +25,39 @@
  */
 final class UserClassBuilder
 {
-    public function addUserInterfaceImplementation(ClassSourceManipulator $manipulator, UserClassConfiguration $userClassConfig)
+    public function addUserInterfaceImplementation(ClassSourceManipulator $manipulator, UserClassConfiguration $userClassConfig): void
     {
         $manipulator->addInterface(UserInterface::class);
 
         $this->addGetUsername($manipulator, $userClassConfig);
 
         $this->addGetRoles($manipulator, $userClassConfig);
 
-        $this->addGetPassword($manipulator, $userClassConfig);
-
-        $this->addGetSalt($manipulator, $userClassConfig);
+        if (06000 < Kernel::VERSION_ID) {
+            $this->addGetPassword($manipulator, $userClassConfig);
+            $this->addGetSalt($manipulator, $userClassConfig);
```

Comment:
In both `getPassword()` and `getSalt()`, when using 5.3 or 5.4, the messaging should change. For `getSalt()` it should have some comment about how the method is deprecated - and you can remove it once using 6.0. For `getPassword()`, the same (unless they are implementing the password interface because they actually need it).

Reply:
Updated the doc blocks for both methods to `This method can be removed in Symfony 6.0 - is not needed for apps that do not check user passwords.` unless the a user is checking passwords.

For `getSalt()`, once https://github.com/symfony/symfony-docs/issues/15065 is completed, I figure we can refactor the docBlocks to point to that doc vs pointing to an interface that may not exist / have a lengthy explanation.